controller/fruitsController.js

```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+
+const router = express.Router();
+// eslint-disable-next-line import/extensions
+const service = require("../services/fruitService.js");
+
+function calculatePrice(price) {
+  const date = new Date();
+  // Price goes up on Fridays
+  if (date.getDay() === 5) {
+    return price + 5;
+  }
+
+  return price;
+}
+
+// GET - return all records
+router.get("/", async (req, res) => {
+  const fruits = await service.getFruits();
+  return res.json(fruits);
+});
+
+// GET - return a specific item
+router.get("/:id", async (req, res) => {
+  const f = service.getFruit(req.params.id);
+
+  if (f === null) {
+    return res.status(404).json({ message: "Fruit not found!" });
+  }
+
+  f.price = calculatePrice(f.price);
+
+  return res.json(f);
+});
+
+// POST - add a new item
+router.post("/", async (req, res) => {
+  // verify inputs
+  if (!req.body.name
+    || !req.body.price.toString().match(/^[0-9]{1,}$/g)) {
```

Comment:
param check should be in seperate helper class

Reply:
Resolved.